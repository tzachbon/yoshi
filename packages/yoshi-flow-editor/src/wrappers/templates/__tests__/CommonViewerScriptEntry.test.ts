import {
  OOI_WIDGET_COMPONENT_TYPE,
  PLATFORM_WIDGET_COMPONENT_TYPE,
} from 'yoshi-flow-editor-runtime/build/constants';
import commonViewerScriptEntry from '../CommonViewerScriptEntry';

describe('CommonViewerScriptEntry template', () => {
  it('generates correct template with a single controller', () => {
    const generateControllerEntryContent = commonViewerScriptEntry({
      viewerScriptWrapperPath:
        'yoshi-flow-editor-runtime/build/viewerScript.js',
      sentryConfig: null,
      projectName: 'someProject',
      defaultTranslations: {},
      biConfig: {},
      visitorBiLoggerPath: 'a/b/c',
      translationsConfig: null,
      appName: 'someapp',
      controllersMeta: [
        {
          controllerFileName: 'project/src/components/button/controller.ts',
          id: '123',
          widgetType: OOI_WIDGET_COMPONENT_TYPE,
          controllerId: 'CONTROLLER_ID',
          componentName: 'SomeWidget',
        },
      ],
      experimentsConfig: {
        scope: 'test-scope',
      },
      viewerEntryFileName: 'project/src/app.ts',
    });

    expect(generateControllerEntryContent).toMatchSnapshot();
  });

  it('generates correct template with a single controller and without viewerEntryFileName', () => {
    const generateControllerEntryContent = commonViewerScriptEntry({
      viewerScriptWrapperPath:
        'yoshi-flow-editor-runtime/build/viewerScript.js',
      sentryConfig: null,
      projectName: 'someProject',
      appName: 'someApp',
      visitorBiLoggerPath: 'a/b/c',
      defaultTranslations: null,
      biConfig: {},
      translationsConfig: {
        default: 'en',
      },
      controllersMeta: [
        {
          controllerFileName: 'project/src/components/button/controller.ts',
          componentName: 'SomeComp',
          id: '123',
          widgetType: OOI_WIDGET_COMPONENT_TYPE,
        },
      ],
      experimentsConfig: {
        scope: 'test-scope',
      },
      viewerEntryFileName: null,
    });

    expect(generateControllerEntryContent).toMatchSnapshot();
  });

  it('generates correct template with multiple controllers', () => {
    const generateControllerEntryContent = commonViewerScriptEntry({
      viewerScriptWrapperPath:
        'yoshi-flow-editor-runtime/build/viewerScript.js',
      sentryConfig: null,
      appName: 'someapp',
      visitorBiLoggerPath: 'a/b/c',
      projectName: 'someProject',
      defaultTranslations: { a: 'b' },
      biConfig: {
        visitor: 'visitor-bi-package',
        owner: 'owner-bi-package',
      },
      translationsConfig: {
        default: 'en',
      },
      controllersMeta: [
        {
          controllerFileName: 'project/src/components/todo/controller.ts',
          id: '123',
          widgetType: OOI_WIDGET_COMPONENT_TYPE,
          controllerId: 'CONTROLLER_ID',
          componentName: 'SomeWidget',
        },
        {
          controllerFileName: 'project/src/components/todo/controller.ts',
          id: '567',
          widgetType: OOI_WIDGET_COMPONENT_TYPE,
          controllerId: 'CONTROLLER_ID',
          componentName: 'SomeWidget',
        },
      ],
      experimentsConfig: {
        scope: 'test-scope',
      },
      viewerEntryFileName: 'project/src/app.ts',
    });

    expect(generateControllerEntryContent).toMatchSnapshot();
  });

  it('generates correct template with controllerId override', () => {
    const generateControllerEntryContent = commonViewerScriptEntry({
      viewerScriptWrapperPath:
        'yoshi-flow-editor-runtime/build/viewerScript.js',
      sentryConfig: null,
      projectName: 'someProject',
      appName: 'someapp',
      defaultTranslations: { a: 'b' },
      visitorBiLoggerPath: null,
      biConfig: {
        visitor: 'visitor-bi-package',
        owner: 'owner-bi-package',
      },
      translationsConfig: {
        default: 'en',
      },
      controllersMeta: [
        {
          controllerFileName: 'project/src/components/todo/controller.ts',
          id: '123',
          widgetType: PLATFORM_WIDGET_COMPONENT_TYPE,
          controllerId: 'CONTROLLER_ID',
          componentName: 'SomeWidget',
        },
        {
          controllerFileName: 'project/src/components/todo/controller.ts',
          id: '567',
          controllerId: '09876',
          widgetType: PLATFORM_WIDGET_COMPONENT_TYPE,
          componentName: 'SomeWidget',
        },
      ],
      experimentsConfig: {
        scope: 'test-scope',
      },
      viewerEntryFileName: 'project/src/app.ts',
    });

    expect(generateControllerEntryContent).toMatchSnapshot();
  });

  it('generates correct template w/o controllers', () => {
    const generateControllerEntryContent = commonViewerScriptEntry({
      viewerScriptWrapperPath:
        'yoshi-flow-editor-runtime/build/viewerScript.js',
      sentryConfig: null,
      projectName: 'someProject',
      visitorBiLoggerPath: 'a/b/c',
      biConfig: {
        visitor: 'visitor-bi-package',
        owner: 'owner-bi-package',
      },
      translationsConfig: {
        default: 'en',
      },
      defaultTranslations: { a: 'b' },
      appName: 'someApp',
      controllersMeta: [],
      experimentsConfig: {
        scope: 'test-scope',
      },
      viewerEntryFileName: 'project/src/app.ts',
    });

    expect(generateControllerEntryContent).toMatchSnapshot();
  });

  it('generates correct template with sentry', () => {
    const generateControllerEntryContent = commonViewerScriptEntry({
      translationsConfig: {
        default: 'en',
      },
      defaultTranslations: { a: 'b' },
      projectName: 'someProject',
      biConfig: {
        visitor: 'visitor-bi-package',
        owner: 'owner-bi-package',
      },
      visitorBiLoggerPath: null,
      viewerScriptWrapperPath:
        'yoshi-flow-editor-runtime/build/viewerScript.js',
      sentryConfig: {
        DSN: 'https:xxx@123',
        teamName: 'team-name',
        projectName: 'project-name',
        id: 'xxx',
      },
      appName: 'someApp',
      controllersMeta: [
        {
          controllerFileName: 'project/src/components/button/controller.ts',
          id: '123',
          widgetType: OOI_WIDGET_COMPONENT_TYPE,
          controllerId: 'CONTROLLER_ID',
          componentName: 'SomeWidget',
        },
      ],
      experimentsConfig: {
        scope: 'test-scope',
      },
      viewerEntryFileName: 'project/src/app.ts',
    });

    expect(generateControllerEntryContent).toMatchSnapshot();
  });
});
