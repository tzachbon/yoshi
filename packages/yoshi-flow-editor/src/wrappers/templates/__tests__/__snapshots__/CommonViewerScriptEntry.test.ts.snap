// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommonViewerScriptEntry template generates correct template w/o controllers 1`] = `
"
  import {createControllersWithDescriptors, initAppForPageWrapper} from 'yoshi-flow-editor-runtime/build/viewerScript.js';
  
  

  
  import * as viewerApp from 'project/src/app.ts';
    var importedApp = viewerApp;


  var sentryConfig = null;

  var experimentsConfig = {\\"scope\\":\\"test-scope\\"};

  var translationsConfig = {\\"default\\":\\"en\\"};

  import biLogger from 'a/b/c';

  export const initAppForPage = initAppForPageWrapper({
    initAppForPage: importedApp.initAppForPage,
    sentry: sentryConfig,
    experimentsConfig: experimentsConfig,
    inEditor: false,
    biLogger: biLogger,
    projectName: \\"someProject\\",
    biConfig: {\\"visitor\\":\\"visitor-bi-package\\",\\"owner\\":\\"owner-bi-package\\"},
    appName: \\"someApp\\",
    translationsConfig: translationsConfig,
  });

  export const createControllers = createControllersWithDescriptors([]);
"
`;

exports[`CommonViewerScriptEntry template generates correct template with a single controller 1`] = `
"
  import {createControllersWithDescriptors, initAppForPageWrapper} from 'yoshi-flow-editor-runtime/build/viewerScript.js';
  
  import controller0 from 'project/src/components/button/controller.ts';

  
  import * as viewerApp from 'project/src/app.ts';
    var importedApp = viewerApp;


  var sentryConfig = null;

  var experimentsConfig = {\\"scope\\":\\"test-scope\\"};

  var translationsConfig = null;

  import biLogger from 'a/b/c';

  export const initAppForPage = initAppForPageWrapper({
    initAppForPage: importedApp.initAppForPage,
    sentry: sentryConfig,
    experimentsConfig: experimentsConfig,
    inEditor: false,
    biLogger: biLogger,
    projectName: \\"someProject\\",
    biConfig: {},
    appName: \\"someapp\\",
    translationsConfig: translationsConfig,
  });

  export const createControllers = createControllersWithDescriptors([{ method: controller0,
          widgetType: \\"WIDGET_OUT_OF_IFRAME\\",
          translationsConfig: null,
          experimentsConfig: {\\"scope\\":\\"test-scope\\"},
          defaultTranslations: {},
          biLogger: biLogger,
          biConfig: {},
          controllerFileName: \\"project/src/components/button/controller.ts\\",
          appName: \\"someapp\\",
          projectName: \\"someProject\\",
          componentName: \\"SomeWidget\\",
          id: \\"CONTROLLER_ID\\" }]);
"
`;

exports[`CommonViewerScriptEntry template generates correct template with a single controller and without viewerEntryFileName 1`] = `
"
  import {createControllersWithDescriptors, initAppForPageWrapper} from 'yoshi-flow-editor-runtime/build/viewerScript.js';
  
  import controller0 from 'project/src/components/button/controller.ts';

  
  var importedApp = {};


  var sentryConfig = null;

  var experimentsConfig = {\\"scope\\":\\"test-scope\\"};

  var translationsConfig = {\\"default\\":\\"en\\"};

  import biLogger from 'a/b/c';

  export const initAppForPage = initAppForPageWrapper({
    initAppForPage: importedApp.initAppForPage,
    sentry: sentryConfig,
    experimentsConfig: experimentsConfig,
    inEditor: false,
    biLogger: biLogger,
    projectName: \\"someProject\\",
    biConfig: {},
    appName: \\"someApp\\",
    translationsConfig: translationsConfig,
  });

  export const createControllers = createControllersWithDescriptors([{ method: controller0,
          widgetType: \\"WIDGET_OUT_OF_IFRAME\\",
          translationsConfig: {\\"default\\":\\"en\\"},
          experimentsConfig: {\\"scope\\":\\"test-scope\\"},
          defaultTranslations: null,
          biLogger: biLogger,
          biConfig: {},
          controllerFileName: \\"project/src/components/button/controller.ts\\",
          appName: \\"someApp\\",
          projectName: \\"someProject\\",
          componentName: \\"SomeComp\\",
          id: \\"123\\" }]);
"
`;

exports[`CommonViewerScriptEntry template generates correct template with controllerId override 1`] = `
"
  import {createControllersWithDescriptors, initAppForPageWrapper} from 'yoshi-flow-editor-runtime/build/viewerScript.js';
  
  import controller0 from 'project/src/components/todo/controller.ts';
import controller1 from 'project/src/components/todo/controller.ts';

  
  import * as viewerApp from 'project/src/app.ts';
    var importedApp = viewerApp;


  var sentryConfig = null;

  var experimentsConfig = {\\"scope\\":\\"test-scope\\"};

  var translationsConfig = {\\"default\\":\\"en\\"};

  var biLogger = null;

  export const initAppForPage = initAppForPageWrapper({
    initAppForPage: importedApp.initAppForPage,
    sentry: sentryConfig,
    experimentsConfig: experimentsConfig,
    inEditor: false,
    biLogger: biLogger,
    projectName: \\"someProject\\",
    biConfig: {\\"visitor\\":\\"visitor-bi-package\\",\\"owner\\":\\"owner-bi-package\\"},
    appName: \\"someapp\\",
    translationsConfig: translationsConfig,
  });

  export const createControllers = createControllersWithDescriptors([{ method: controller0,
          widgetType: \\"STUDIO_WIDGET\\",
          translationsConfig: {\\"default\\":\\"en\\"},
          experimentsConfig: {\\"scope\\":\\"test-scope\\"},
          defaultTranslations: {\\"a\\":\\"b\\"},
          biLogger: biLogger,
          biConfig: {\\"visitor\\":\\"visitor-bi-package\\",\\"owner\\":\\"owner-bi-package\\"},
          controllerFileName: \\"project/src/components/todo/controller.ts\\",
          appName: \\"someapp\\",
          projectName: \\"someProject\\",
          componentName: \\"SomeWidget\\",
          id: \\"CONTROLLER_ID\\" }, { method: controller1,
          widgetType: \\"STUDIO_WIDGET\\",
          translationsConfig: {\\"default\\":\\"en\\"},
          experimentsConfig: {\\"scope\\":\\"test-scope\\"},
          defaultTranslations: {\\"a\\":\\"b\\"},
          biLogger: biLogger,
          biConfig: {\\"visitor\\":\\"visitor-bi-package\\",\\"owner\\":\\"owner-bi-package\\"},
          controllerFileName: \\"project/src/components/todo/controller.ts\\",
          appName: \\"someapp\\",
          projectName: \\"someProject\\",
          componentName: \\"SomeWidget\\",
          id: \\"09876\\" }]);
"
`;

exports[`CommonViewerScriptEntry template generates correct template with multiple controllers 1`] = `
"
  import {createControllersWithDescriptors, initAppForPageWrapper} from 'yoshi-flow-editor-runtime/build/viewerScript.js';
  
  import controller0 from 'project/src/components/todo/controller.ts';
import controller1 from 'project/src/components/todo/controller.ts';

  
  import * as viewerApp from 'project/src/app.ts';
    var importedApp = viewerApp;


  var sentryConfig = null;

  var experimentsConfig = {\\"scope\\":\\"test-scope\\"};

  var translationsConfig = {\\"default\\":\\"en\\"};

  import biLogger from 'a/b/c';

  export const initAppForPage = initAppForPageWrapper({
    initAppForPage: importedApp.initAppForPage,
    sentry: sentryConfig,
    experimentsConfig: experimentsConfig,
    inEditor: false,
    biLogger: biLogger,
    projectName: \\"someProject\\",
    biConfig: {\\"visitor\\":\\"visitor-bi-package\\",\\"owner\\":\\"owner-bi-package\\"},
    appName: \\"someapp\\",
    translationsConfig: translationsConfig,
  });

  export const createControllers = createControllersWithDescriptors([{ method: controller0,
          widgetType: \\"WIDGET_OUT_OF_IFRAME\\",
          translationsConfig: {\\"default\\":\\"en\\"},
          experimentsConfig: {\\"scope\\":\\"test-scope\\"},
          defaultTranslations: {\\"a\\":\\"b\\"},
          biLogger: biLogger,
          biConfig: {\\"visitor\\":\\"visitor-bi-package\\",\\"owner\\":\\"owner-bi-package\\"},
          controllerFileName: \\"project/src/components/todo/controller.ts\\",
          appName: \\"someapp\\",
          projectName: \\"someProject\\",
          componentName: \\"SomeWidget\\",
          id: \\"CONTROLLER_ID\\" }, { method: controller1,
          widgetType: \\"WIDGET_OUT_OF_IFRAME\\",
          translationsConfig: {\\"default\\":\\"en\\"},
          experimentsConfig: {\\"scope\\":\\"test-scope\\"},
          defaultTranslations: {\\"a\\":\\"b\\"},
          biLogger: biLogger,
          biConfig: {\\"visitor\\":\\"visitor-bi-package\\",\\"owner\\":\\"owner-bi-package\\"},
          controllerFileName: \\"project/src/components/todo/controller.ts\\",
          appName: \\"someapp\\",
          projectName: \\"someProject\\",
          componentName: \\"SomeWidget\\",
          id: \\"CONTROLLER_ID\\" }]);
"
`;

exports[`CommonViewerScriptEntry template generates correct template with sentry 1`] = `
"
  import {createControllersWithDescriptors, initAppForPageWrapper} from 'yoshi-flow-editor-runtime/build/viewerScript.js';
  
  import controller0 from 'project/src/components/button/controller.ts';

  
  import * as viewerApp from 'project/src/app.ts';
    var importedApp = viewerApp;


  var sentryConfig = {
      DSN: 'https:xxx@123',
      id: 'xxx',
      projectName: 'project-name',
      teamName: 'team-name',
    };

  var experimentsConfig = {\\"scope\\":\\"test-scope\\"};

  var translationsConfig = {\\"default\\":\\"en\\"};

  var biLogger = null;

  export const initAppForPage = initAppForPageWrapper({
    initAppForPage: importedApp.initAppForPage,
    sentry: sentryConfig,
    experimentsConfig: experimentsConfig,
    inEditor: false,
    biLogger: biLogger,
    projectName: \\"someProject\\",
    biConfig: {\\"visitor\\":\\"visitor-bi-package\\",\\"owner\\":\\"owner-bi-package\\"},
    appName: \\"someApp\\",
    translationsConfig: translationsConfig,
  });

  export const createControllers = createControllersWithDescriptors([{ method: controller0,
          widgetType: \\"WIDGET_OUT_OF_IFRAME\\",
          translationsConfig: {\\"default\\":\\"en\\"},
          experimentsConfig: {\\"scope\\":\\"test-scope\\"},
          defaultTranslations: {\\"a\\":\\"b\\"},
          biLogger: biLogger,
          biConfig: {\\"visitor\\":\\"visitor-bi-package\\",\\"owner\\":\\"owner-bi-package\\"},
          controllerFileName: \\"project/src/components/button/controller.ts\\",
          appName: \\"someApp\\",
          projectName: \\"someProject\\",
          componentName: \\"SomeWidget\\",
          id: \\"CONTROLLER_ID\\" }]);
"
`;
